src/reyna/HttpClient.cs

```diff
@@ -54,24 +56,46 @@ public Result Post(IMessage message)
 
         internal static Result CanSend()
         {
-            ConnectionInfo connectionInfo = new ConnectionInfo();
-            if (!connectionInfo.Connected)
+            ConnectionInfo info = new ConnectionInfo();
+            if (!info.Connected)
             {
                 return Result.NotConnected;
             }
 
-            TimeRange range = Preferences.CellularDataBlackout;
-            if (range == null)
+            if (Preferences.WwanBlackoutRange == null)
             {
-                return Result.Ok;
+                SaveCellularDataAsWwanForBackwardsCompatibility();
             }
 
-            if (!connectionInfo.Mobile)
+            PowerManager powerManager = new PowerManager();
+            if (powerManager.IsBatteryCharging() && !Preferences.OnChargeBlackout)
```

Comment:
That might also be based on a old Android version. We should not invert for Preferences.OnChargeBlackout here, if blackout is true we should blackout and not send.